synapse/http/server.py

```diff
@@ -275,12 +277,9 @@ def render(self, request):
             This checks if anyone has registered a callback for that method and
             path.
         """
-        callback, group_dict = self._get_handler_for_request(request)
+        callback, servlet_classname, group_dict = self._get_handler_for_request(request)
 
-        servlet_instance = getattr(callback, "__self__", None)
-        if servlet_instance is not None:
-            servlet_classname = servlet_instance.__class__.__name__
-        else:
+        if servlet_classname is None:
```

Comment:
when would it be None?

Reply:
I'm just defending against the event that someone passes None for the classname. We still need a unique name to reference it in prometheus.

Reply:
I've added a comment to that effect.

Reply:
If we need to defend against that, I think I'd rather see it done in `register_paths`: however, I don't think it's a thing we need to defend against?

Reply:
To be honest I just threw it in because I had the same confusion from line 281 in the old code. Why was there the test `if servlet_instance is not None:`? I'll remove it.

Reply:
It's not necessary. I've removed it.

Reply:
> Why was there the test `if servlet_instance is not None`:

because sometimes the `callback` would be a regular function rather than an instance method (for example: `_options_handler`), so would have no `__self__` attribute. That's no longer a concern.

Reply:
Aha, that makes sense